### PR TITLE
Add hex codes

### PR DESCRIPTION
**Pull request by MeestorX**
Typing %hh to send values. E.g. %0A sends a CR

Fixed a couple of other typos.

### CODE REVIEW

**Comment by krocheck** [comment]
`unescape()` is technically a deprecated method.  Is there another way to accomplish this @MeestorX ?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/unescape

**Comment by MeestorX** [comment]
> `unescape()` is technically a deprecated method. Is there another way to accomplish this @MeestorX ?
> 
> https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/unescape

I did a lot of searching to find this one. Couldn't really find anything else that could be "plugged in", but I'm just learning JS so likely there's a better way, I just don't know what it would be, short of writing a bunch of code to parse the line manually?

**Comment by krocheck** [comment]
@Julusian or @haakonnessjoen thoughts on using this function?

**Comment by JeffreyDavidsz** [comment]
That can be a regex function.

**Comment by MeestorX** [comment]
Love to see how that's done. Please enlighten me if you get a chance.

**Comment by JeffreyDavidsz** [comment]
Something like this?
`
function unescape (obj, escaped, char) {
	for (const key in obj) {
			// Replace the escaped version with the character
			obj[key] = obj[key].replace(new RegExp(escaped, "g"), char);
	}
}
`
`unescape(object,"&#x2F;", "/")`

**Comment by MeestorX** [comment]
I tried doing a replace and it didn't work. It needed each escaped hex value to be replaced by the character that hex represented.

e.g. "%0D" needs to be replaced by "\x0D" not just the % replaced by a "\x".

Maybe your code addresses that?

**Comment by JeffreyDavidsz** [comment]
Its just an example it converts the HEX value of /(&#x2F;) to / again for display. but you can do it however you want

**Comment by JeffreyDavidsz** [comment]
put in a regex instead of string and it will replace all

**Comment by JeffreyDavidsz** [comment]
mmm crap sorry misread. You want to convert also -in one function-

**Comment by MeestorX** [comment]
Yeah, spent a lot of time getting nowhere on this. unescape() worked and was the easiest solution.

File: index.js [hunk 1]
```diff
@@ -196,12 +196,12 @@ instance.prototype.destroy = function() {
...
 instance.prototype.CHOICES_END = [
-	{ id: '', 			label: 'None' },
-	{ id: '\n', 		label: 'CR - \\n (Common UNIX/Mac)' },
-	{ id: '\r\n', 	label: 'CRLF - \\r\\n (Common Windows)' },
-	{ id: '\r', 		label: 'LF - \\r (Old MacOS)' },
-	{ id: '\x00', 	label: 'NULL - \\x00 (Can happen)' },
-	{ id: '\n\r', 	label: 'LFCR - \\n\\r (Just stupid)' },
+	{ id: '',     label: 'None' },
+	{ id: '\n',   label: 'CR - \\n (Common UNIX/Mac)' },
+	{ id: '\n\r', label: 'CRLF - \\n\\r (Common Windows)' },
```
Comment (haakonnessjoen) [inline index.js:201]: Wrong CR LF is \r \n, why did you change this?

File: index.js [hunk 2]
```diff
@@ -196,12 +196,12 @@ instance.prototype.destroy = function() {
...
+	{ id: '\r',   label: 'LF - \\r (Old MacOS)' },
+	{ id: '\x00', label: 'NULL - \\x00 (Can happen)' },
+	{ id: '\r\n', label: 'LFCR - \\r\\n (Just stupid)' },
```
Comment (haakonnessjoen) [inline index.js:204]: Wrong

**Comment by haakonnessjoen** [comment]
What, didn't see that this is already merged. Could you please fix this @JeffreyDavidsz ?

**Comment by MeestorX** [comment]
@haakonnessjoen I was told it was wrong originally, so I "fixed" it, but the wrong lines! Oops!
Has it been corrected?

**Comment by JeffreyDavidsz** [comment]
mmm "new changes since you last viewed" and then, merge, oeps.
I've changed it to the correct settings, although the others where wrong to. CR != \n for example (as you pointed out for the windows example). When I got the chance, I'll upgrade into the core.
@MeestorX did you found the right solution for the hex stuff? I thing you need something like 
`
let foo = buf.toString('hex'); //convert a buffer to string 
new Buffer.from(foo, 'hex') ; //convert a string to buffer
`

**Comment by MeestorX** [comment]
I’m open to suggestions. As I say, I spent a lot of time getting nowhere so just used unescape() because it worked.

**Comment by haakonnessjoen** [comment]
Yeah, unescape seems like a nice solution. I wouldn't want to change that. But just change back the line endings :)

```console.log(unescape('hey%20Ho%0a100%'));
hey Ho
100%```

**Comment by haakonnessjoen** [comment]
Ah yes, Jeffrey has already fixed them :)

**Comment by jgwhitebeard** [comment]
Hi all. Great new feature. Thanks. It looks like the unescape approach limits us to 7 bit characters though so it can't send any value greater than \x79 without splitting it into 2 bytes. Makes sending a final \xFF impossible which rules out a few protocols! Any ideas?

**Comment by istnv** [comment]
Pull request #2 adds a patch to avoid the utf8 corruption.
